app/src/main/java/app/revanced/manager/ui/component/settings/BooleanItem.kt

```diff
@@ -5,27 +5,53 @@ import androidx.compose.foundation.clickable
 import androidx.compose.material3.Switch
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
 import app.revanced.manager.domain.manager.base.Preference
+import app.revanced.manager.ui.component.SafeguardConfirmationDialog
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
 
 @Composable
 fun BooleanItem(
     modifier: Modifier = Modifier,
+    isSafeguard: Boolean = false,
```

Comment:
This parameter can be removed if you rename the overload to `SafeguardBooleanItem` and copy the original function from the `compose-dev` branch